### PR TITLE
Add benchmarks to MHLO miniLM and resnet50 and fix some issues with tolerances for TF32.

### PR DESCRIPTION
**Pull request by monorimet**
NOTE: benchmark reference numbers from TF on GPU are showing CPU results as GPUs are disabled in tf.config.
It's a bit tricky to navigate TF GPU execution/visibility so that will come in a follow-up.

### CODE REVIEW

File: tank/model_utils_tf.py [hunk 1]
```diff
@@ -140,6 +210,8 @@ def get_causal_lm_model(hf_name, text="Hello, this is the default text."):
     #    gpus = tf.config.experimental.list_physical_devices("GPU")
     #    for gpu in gpus:
     #        tf.config.experimental.set_memory_growth(gpu, True)
+    BATCH_SIZE = 1
+    MAX_SEQUENCE_LENGTH = 16
```
Comment (powderluv) [inline tank/model_utils_tf.py:214]: We want 128

File: shark/shark_benchmark_runner.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def benchmark_torch(self, modelname):
         torch_device = torch.device(
             "cuda:0" if self.device == "gpu" else "cpu"
         )
-        HFmodel, input, act_out = get_torch_model(modelname)
+        HFmodel, input, _ = get_torch_model(modelname)
```
Comment (dan-garvey) [inline shark/shark_benchmark_runner.py:79]: If you know what this is, even if it doesn't get used, its better to give it a name, or not assign it and append "[:2]" to get_torch_model(modelname) (assuming it returns a list or tuple)

File: shark/shark_benchmark_runner.py [hunk 3]
```diff
@@ -98,26 +98,32 @@ def benchmark_torch(self, modelname):
...
-        # return [
-        #    f"{shark_args.num_iterations/(end-begin)}",
-        #    f"{((end-begin)/shark_args.num_iterations)*1000}",
-        # ]
-        return ["n/a", "n/a"]
+    def benchmark_tf(self, modelname):
+        import tensorflow as tf
+        from tank.model_utils_tf import get_tf_model
+
+        model, input, _ = get_tf_model(modelname)
```
Comment (dan-garvey) [inline shark/shark_benchmark_runner.py:105]: same as above

File: tank/MiniLM-L12-H384-uncased_torch/MiniLM-L12-H384-uncased_torch_test.py [hunk 4]
```diff
@@ -19,17 +20,7 @@ def create_and_check_module(self, dynamic, device):
...
-        # mlir_importer = SharkImporter(
-        #    model,
-        #    (input,),
-        #    frontend="torch",
-        # )
-        # minilm_mlir, func_name = mlir_importer.import_mlir(
-        #    is_dynamic=dynamic, tracing_required=True
-        # )
-
+        # shark_args.enable_tf32 = self.benchmark
```
Comment (dan-garvey) [inline tank/MiniLM-L12-H384-uncased_torch/MiniLM-L12-H384-uncased_torch_test.py:23]: Delete or add a comment why it is commented and a TODO: with what needs to be done

File: tank/model_utils_tf.py [hunk 5]
```diff
@@ -16,10 +16,48 @@
...
+    "mpnet-base",
+    "rembert",
+    "roberta-base",
+    "tapas-base",
+    "tiny-random-flaubert",
+    "xlm-roberta",
+]
+tfhf_models = [
+    "microsoft/MiniLM-L12-H384-uncased",
+]
```
Comment (dan-garvey) [inline tank/model_utils_tf.py:44]: Maintaining this list seems irritating, is there a way it could be generated? Can be a TODO

File: tank/model_utils_tf.py [hunk 6]
```diff
@@ -57,7 +95,7 @@ def get_TFhf_model(name):
         text,
         padding="max_length",
         truncation=True,
-        max_length=MAX_SEQUENCE_LENGTH,
+        max_length=128,
```
Comment (dan-garvey) [inline tank/model_utils_tf.py:98]: Why is this changed when MAX_SEQUENCE_LENGTH still is 128? Ideally don't use numbers when you can use a named variable

**Review by dan-garvey — CHANGES_REQUESTED** [review]
use git commit --amend instead of making 7 commits. Squash those so commit history is cleaner = )

File: tank/model_utils_tf.py (hunk 5 above)
Comment (monorimet) [reply tank/model_utils_tf.py]: yes -- maybe we can pull it from tank/tf/tf_model_list.csv
I have added a TODO.

File: tank/model_utils_tf.py (hunk 6 above)
Comment (monorimet) [reply tank/model_utils_tf.py]: This was a temporary fix to a problem that I have fixed in model_utils_tf. See new commit.

Comment (powderluv) [reply tank/model_utils_tf.py]: You can still set the #define to 128. In general numbers in code should be #defines

**Review by powderluv — DISMISSED** [review]

File: shark/shark_benchmark_runner.py [hunk 7]
```diff
@@ -58,16 +57,17 @@ def setup_cl(self, input_tensors):
             input_tensors,
             mlir_dialect=self.mlir_dialect,
         )
-        print(self.benchmark_cl)
+        # print(self.benchmark_cl)
```
Comment (dan-garvey) [inline shark/shark_benchmark_runner.py:60]: dead

**Review by dan-garvey — APPROVED** [review]
+1

File: shark/shark_benchmark_runner.py (hunk 7 above)
Comment (dan-garvey) [reply shark/shark_benchmark_runner.py:60]: (sorry if I missed this on the first pass)
